internal/controller/controllers/cluster_controller.go

```diff
@@ -323,6 +331,23 @@ func (r *ClusterReconciler) updateIfNeeded(ctx context.Context, cluster *adiiov1
 		return update, ctrl.Result{}, errors.Wrap(err, "failed to update cluster")
 	}
 
+	if pullSecretUpdate {
+		images := &adiiov1alpha1.ImageList{}
```

Comment:
please move this part to a helper function.

Reply:
ok